### PR TITLE
Player notes update

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Updates out Notes system so it's database based and the same as /tg/'s (by Jordie0608): https://github.com/tgstation/-tg-station/pull/11222.
2) Adds the Bygex system from /tg/, which allows for Regex in BYOND.

Our notes system isn't too great. All you can do is add notes and remove your own notes, and you can't edit them. Additionally, they're saved in a local file on the server that probably hasn't been backed up in two years now. This change converts the notes system into one that's more similar to the watchlist, which also makes it database based.

Added functionality:
* Search for player notes.
* Remove notes.
* Edit notes.
* Log of note edits.

Removed functionality:
* None! All the verbs are the same!

The main downside of this would be the interface, which isn't too pretty but is something we can get used to. In the end I'll need admin input on this though, as they're the ones mostly using the system (see the screenshot below).

*Requires an SQL change and config change prior to merging.*

### CODE REVIEW

**Comment by Nomzosha** [comment]
:+1:  looks cool

File: code/modules/admin/sql_notes.dm [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+			var/err = query_list_notes.ErrorMsg()
+			log_game("SQL ERROR obtaining ckey from notes table. Error : \[[err]\]\n")
+			return
+		while(query_list_notes.NextRow())
+			index_ckey = query_list_notes.item[1]
+			output += "<a href='?_src_=holder;shownoteckey=[index_ckey]'>[index_ckey]</a><br>"
+	else
+		output += "<center><a href='?_src_=holder;addnoteempty=1'>\[Add Note\]</a></center>"
+		output += ruler
+	usr << browse(output, "window=show_notes;size=900x500")
```
Comment (tigercat2000) [inline code/modules/admin/sql_notes.dm]: You might try using the /datum/browser stuff for this, it tends to make things look nice even with weird HTML.

**Comment by Fox-McCloud** [comment]
All the yes.

:+1:  x :100:

**Comment by tigercat2000** [comment]
I'm firmly eh about this. Like, I don't think notes really need to be any more complicated.

**Comment by FalseIncarnate** [comment]
The ability to edit the notes alone is worth a :+1: from me, really sucks when you accidentally mess up the spelling or wording and have to make a completely new note to correct it.

**Comment by Fox-McCloud** [comment]
For me, the ability to search for notes is incredibly useful, instead of manually searching for them, alphabetically.

**Comment by Markolie** [comment]
Turns out our old notes system was completely different from /tg/'s, so getting the converter to function was problematic. Should work a-ok now though.

**Comment by Earthdivine** [comment]
My only concern is that when we changed the Watchlist, it didn't save timestamps and people who put them on there. Will the notes end up like that too?

Also yea, the ability to remove and edit note is a big :+1: , Though will note removal be tracked in some way?

**Comment by Markolie** [comment]
Note removal is logged, and it also logs what the note used to say.

It will save all original data of the old notes.

**Comment by Earthdivine** [comment]
Awsome.

**Comment by Markolie** [comment]
Okay, the mass conversion proc has been tested on 3000+ notes, so it should be good to go. The plan to roll this out:

1) I'll contact Earth for the SQL changes.
2) The server is patched.
3) We call the mass_convert_notes() on the live server, which'll cause it to hang for a few minutes while it converts all notes.

**Pull request merged**